src/directory/managed_directory.rs

```diff
@@ -65,53 +94,98 @@ impl ManagedDirectory {
     /// an error is simply logged, and the file remains in the list of managed
     /// files.
     pub fn garbage_collect(&mut self, living_files: HashSet<PathBuf>) {
-        let mut managed_has_changed: bool = false;
-        {
-            let mut files_to_delete = vec!();
-            let mut managed_paths_write = self.managed_paths.write().unwrap();
+        let mut files_to_delete = vec!();
+        {   // releasing the lock as .delete() will use it too.
+            let mut meta_informations_wlock = self.meta_informations.write().unwrap();
```

Comment:
expect(...)

double check that expect is ok here.
ie: the lock is released right away, with code that cannot panick (no io, no call to other library etc.)